src/makeCalculator.js

```diff
@@ -4,7 +4,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    operate(func, num) {
+      this.result = func(this.result, num);
```

Comment:
The `operate` method should check if `func` is a function before calling it. You can use `typeof func === 'function'` to ensure that the provided callback is indeed a function. This will prevent runtime errors if a non-function is passed.